docs/tasks/argo.rst

```diff
@@ -0,0 +1,253 @@
+------------------------------------------------------------
+
+Argo Workflows enable us to schedule operations. In the Kanister project,
+Argo Cron Workflows will be used to automate the creation of ActionSets to
+backup an application at regular intervals.
+
+To summarize, ActionSets are CRDs that are used to backup or restore an application.
+The controller present in the Kanister namespace looks for ActionSets to
+execute in the given namespace.
+
+In this tutorial you will schedule the creation of a backup actionset using
+Argo Cron Workflows.
+
+
+Prerequisites
+===============
+- Kubernetes 1.16+
+- Kanister controller version 0.80.0 installed in your cluster, let's assume in
+  Namespace kanister
+- Kanctl CLI installed (https://docs.kanister.io/tooling.html#install-the-tools)
+
+Architecture
+===============
+
+.. image:: img/argo-cron-architecture.png
+
+Steps -
+===============
+
+
+Step 1 - Setup Argo
+``````````````````````
+
+Download the Argo CLI from their Releases_ page.
+
+Create a separate namespace for the Workflows.
+
+.. code-block:: bash
+
+  kubectl create ns argo
+
+For this tutorial, the minimal manifest file will be used to download
+Argo CRDs and setup their resources.
+Download the minimal manifest file from their Examples_ page.
+
+On downloading this yaml file, you may configure the managed namespaces
+as per your use case.
+You may read more about managed namespaces here - ManagedNamespaces_
+
+In this tutorial, the minimal manifest will be used as it is.
+
+.. code-block:: bash
+
+  kubectl apply -f quick-start-minimal.yaml -n argo
+
+Port-forward to view the Argo UI:
+
+.. code-block:: bash
+
+  kubectl -n argo port-forward deployment/argo-server 2746:2746
+
+Open a web browser and navigate to ``localhost:2746``
+
+.. image:: img/argo-default-ui.png
+
+Step 2 - Setup a sample application to backup.
+``````````````````````````````````````````````````
+
+Here, you will reference the MySQL_ example from Kanister.
+Install the Chart and setup MySQL in the ``mysql-test`` namespace.
+Make sure that the application is integrated with Kanister and the following
+steps are completed.
+
+1. A Profile CR is created for the ``mysql-test`` namespace.
+2. A MySQL blueprint is created in the ``kanister`` namespace.
+3. Check the name of the blueprint, stateful set, profile and
+   secrets for you application for the next step.
+
+Step 3 - Creating a Cron Workflow
+````````````````````````````````````````````
+
+Now a Cron Workflow will be created that will automate the creation of an actionset
+to backup our MySQL application. You will need to modify the names of the blueprint,
+stateful set, profile and secrets as per the ones you have created.
+Modify the last line of the following yaml file as follows.
+
+.. code-block:: bash
+
+  kanctl create actionset --action backup --namespace kanister --blueprint [YOUR_BLUEPRINT_NAME] --statefulset [YOUR_STATEFULSET] --profile [YOUR_PROFILE_NAME] --secrets [YOUR_SECRETS_NAME]
+
```

Comment:
I think instead of mentioning this command separately we can directly mention this in the CronWorkflow manifest.

Reply:
I wanted to keep the difference between following the MySQL example and using their own configuration. If the MySQL example is followed word-by-word, only the name of the Profile will change. The rest of the command remains same. Replacing this will also create confusion as to where each of these resources is located because we mention namespace/resource-name for some of the resources.

Reply:
so, in the above command they still have to specify the namespace and then statefulname and similalry for secret and profile, right?

Reply:
If they do not follow the example, yes. But from what I have heard, the creation of an ActionSet can be even more customizable. So the number and type of arguments passed can be different.

Reply:
Done.